kale/utils/metadata_utils.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from kale.utils.pod_utils import is_workspace_dir
+
+METADATA_REQUIRED_KEYS = [
+    'experiment_name',
+    'pipeline_name',
+]
+kale_step_name_regex = r'^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
+kale_name_msg = ("must consist of lower case alphanumeric characters "
+                 "or '-', and must start and end with an alphanumeric"
+                 " character.")
+k8s_valid_name_regex = r'^[\.\-a-z0-9]+$'
+k8s_name_msg = ("must consist of lower case alphanumeric characters, "
+                "'-' or '.'")
+volume_types = ['pv', 'pvc', 'new_pvc', 'clone']
+volume_required_fields = ['name', 'annotations', 'size', 'type', 'mount_point']
```

Comment:
Module constants are typically named in all caps (https://www.python.org/dev/peps/pep-0008/#constants). Also, you may want to define these as tuples, which are immutable.